app/transport/rest/server.go

```diff
@@ -49,4 +49,11 @@ func (srv *Server) Run() error {
 func attachUserEndpoints(router *mux.Router, handlers Handlers) {
 	router.Path("/users").Methods(http.MethodPost).Handler(handlers.UserHandler.CreateUser())
 	router.Path("/users/{id}").Methods(http.MethodGet).Handler(handlers.UserHandler.GetUserByID())
+
+	var allowedSortArgs = strings.Join([]string{firstName, lastName, createdAt}, "|")
+	router.Path("/users").Methods(http.MethodGet).
+		Queries(offset, "{"+offset+":[0-9]+}").
```

Comment:
Is these `Queries(offset, "{"+offset+":[0-9]+}")` are required?
I'm not familiar with it but looks strange. I don't want to specify this when i'm declaring the handler.

Reply:
We have to set up rules for query parameters otherwise we will catch panic here in case of a bad URI.  We can remove this code and improve it with middleware that would be responsible for validation and error handling.

Reply:
Our endpoint would not work properly if we do not set up rules or make query validation. Maybe middleware will address the problem.